src/App.tsx

```diff
@@ -1,61 +1,144 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import { useState } from 'react';
+
+interface ServerTodo {
+  completed: boolean;
+  id: number;
+  title: string;
+  userId: number;
+}
+
+function getUserFromId(todo: ServerTodo): User {
+  return usersFromServer.find(user => user.id === todo.userId) as User;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserFromId(todo),
+}));
+
+const regex = /^[a-zA-Za-яА-ЯiІіЇїґҐєЄ0-9\s]+$/;
 
 export const App = () => {
+  const [todosArray, setTodosArray] = useState<Todo[]>(preparedTodos);
+
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [hasSelectedUserError, setHasSelectedUserError] = useState(false);
+  const [isDefaultDisabled, setIsDefaultDisabled] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const handleSelectedUserError = () => {
+    if (selectedUser === 0) {
+      setHasSelectedUserError(true);
+    }
+  };
+
+  const handleHasTitleError = () => {
+    if (title === '') {
```

Comment:
```suggestion
    if (!title) {
```
Don't compare with an empty string, check if a value exists